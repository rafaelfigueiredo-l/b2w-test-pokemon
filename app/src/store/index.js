import { createStore } from "redux";

const INITIAL_STATE = {
  theme: {},
  pokemon: [],
};
function course(state = INITIAL_STATE, action) {
  switch (action.type) {
    case "SET_THEME":
      return {
        ...state,
        theme: action.payload,
      };

    case "SET_POKEMON":
      return {
        ...state,
        pokemon: action.payload,
      };

    default:
      return state;
  }
}

const store = createStore(
  course,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);

export default store;
