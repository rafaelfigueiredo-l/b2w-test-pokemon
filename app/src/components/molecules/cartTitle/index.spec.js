import { Provider } from "react-redux";
import store from "../../../store";
import CartTitle from "./";

describe("<CartTitle />", () => {
  const component = (
    <Provider store={store}>
      <CartTitle />
    </Provider>
  );

  const mounted = mount(component);

  it("should mount correctly CartTitle component", () => {
    expect(mounted.find(".cartTitle")).toHaveLength(2);
    expect(mounted).toMatchSnapshot();
  });

  it("should unmount correctly CartTitle component", () => {
    expect(mounted.unmount()).toEqual({});
  });
});
