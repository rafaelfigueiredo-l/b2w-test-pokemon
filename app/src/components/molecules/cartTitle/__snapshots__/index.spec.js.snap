// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CartTitle /> should mount correctly CartTitle component 1`] = `
<styled.div
  className="cartTitle"
>
  <div
    className="sc-AxirZ fUMLcy cartTitle"
  >
    <styled.span
      className="cartText"
    >
      <span
        className="sc-AxjAm eKjPIR cartText"
      >
        Carrinho
      </span>
    </styled.span>
  </div>
</styled.div>
`;
